cinema/views.py

```diff
@@ -43,6 +51,26 @@ def get_serializer_class(self):
 
         return MovieSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action in ("list", "retrieve"):
+            queryset = queryset.prefetch_related("genres", "actors")
+
+        title = self.request.query_params.get("title")
+        genres = self.request.query_params.get("genres")
+        actors = self.request.query_params.get("actors")
+
+        if title:
+            queryset = queryset.filter(title__icontains=title)
+        if genres:
+            genres = self._params_to_ints(genres)
+            queryset = queryset.filter(genres__id__in=genres)
+        if actors:
+            actors = self._params_to_ints(actors)
```

Comment:
Consider adding error handling for the `_params_to_ints` method to manage cases where the query string contains non-integer values. This will prevent potential runtime errors when filtering by genres or actors.